src/Core/Client/RetryMiddleware.php

```
@@ -0,0 +1,191 @@
+     *     maxRetries: int,
+     * } $options
+     */
+    private function onFulfilled(RequestInterface $request, array $options): callable
+    {
+        $retryAttempt = $options['retryAttempt'];
+        $maxRetries = $options['maxRetries'];
+        return function ($value) use ($request, $options, $retryAttempt, $maxRetries) {
+            if (!$this->shouldRetry(
+                $retryAttempt,
+                $maxRetries,
+                $value
```

PHPStan (2.1.17)

Parameter #3 $response of method Intercom\Core\Client\RetryMiddleware::shouldRetry() expects Psr\Http\Message\ResponseInterface|null, mixed given.


(argument.type)